src/tools/miri/src/machine.rs

```diff
@@ -552,6 +565,8 @@ pub struct MiriMachine<'mir, 'tcx> {
     /// The spans we will use to report where an allocation was created and deallocated in
     /// diagnostics.
     pub(crate) allocation_spans: RefCell<FxHashMap<AllocId, (Span, Option<Span>)>>,
+
+    const_cache: RefCell<FxHashMap<(mir::Const<'tcx>, usize), OpTy<'tcx, Provenance>>>,
```

Comment:
```suggestion
    /// Maps MIR consts to their evaluated result. We combine the const with a "salt" (`usize`)
    /// that is fixed per stack frame; this lets us have sometimes different results for the
    /// same const while ensuring consistent results within a single call.
    const_cache: RefCell<FxHashMap<(mir::Const<'tcx>, usize), OpTy<'tcx, Provenance>>>,
```

Reply:
_ping_
GitHub doesn't say that this suggestion is outdated, but I have indeed applied it.